eclair-core/src/main/scala/fr/acinq/eclair/db/Databases.scala

```diff
@@ -162,6 +174,71 @@ object Databases extends Logging {
         }
       }
 
+      safetyChecks_opt foreach { initChecks =>
+
+        PgUtils.inTransaction { connection =>
+          using(connection.createStatement()) { statement =>
+
+            def checkMaxAge(name: String, maxAge: FiniteDuration, sqlQuery: String): Unit = {
+              import ExtendedResultSet._
+              val smallestAge_opt = statement
+                .executeQuery(sqlQuery)
+                .headOption // sql max() will always return a result, with perhaps a null value if there was no records
+                .flatMap(_.getTimestampNullable("max"))
+                .map(ts => TimestampMilli.now() - TimestampMilli.fromSqlTimestamp(ts))
+              require(smallestAge_opt.isDefined, s"db check failed: no $name found")
+              require(smallestAge_opt.get <= maxAge, s"db check failed: most recent $name is too old (${smallestAge_opt.get.toMinutes} minutes > ${maxAge.toMinutes} minutes)")
+              logger.info(s"db check ok: max age ${smallestAge_opt.get.toMinutes} minutes <= ${maxAge.toMinutes} minutes for $name")
+            }
+
+            checkMaxAge(name = "local channel",
+              maxAge = initChecks.localChannelsMaxAge,
+              sqlQuery =
+                """
+                  |SELECT MAX(GREATEST(created_timestamp, last_payment_sent_timestamp, last_payment_received_timestamp, last_connected_timestamp, closed_timestamp))
+                  |FROM local.channels
+                  |WHERE NOT is_closed""".stripMargin)
+
+            checkMaxAge(name = "network node",
+              maxAge = initChecks.networkNodesMaxAge,
+              sqlQuery =
+                """
+                  |SELECT MAX((json->'timestamp'->>'iso')::timestamptz)
+                  |FROM network.nodes""".stripMargin)
+
+            checkMaxAge(name = "audit relayed",
+              maxAge = initChecks.auditRelayedMaxAge,
+              sqlQuery =
+                """
+                  |SELECT MAX(timestamp)
+                  |FROM audit.relayed""".stripMargin)
+
+            def checkMinCount(name: String, minCount: Int, sqlQuery: String): Unit = {
+              import ExtendedResultSet._
+              val count = statement
+                .executeQuery(sqlQuery)
+                .map(_.getInt("count"))
+                .head
```

Comment:
nit: add a comment here saying that `COUNT(*)` will always return one row?

Reply:
Done with 4ebd783e3.